src/Kubernetes.Controller/Caching/NamespaceCache.cs

```diff
@@ -149,6 +151,37 @@ public ImmutableList<string> Update(WatchEventType eventType, V1Service service)
         }
     }
 
+    public ImmutableList<string> Update(WatchEventType eventType, V1Pod pod)
+    {
+        if (pod is null)
+        {
+            throw new ArgumentNullException(nameof(pod));
+        }
+
+        var podName = pod.Name();
+        lock (_sync)
```

Comment:
not sure how the C# SDK works, but in Go you get the Watch events from a sequentially processed list, so no need for locks.

Reply:
This is the pattern used elsewhere in the project.

Reply:
Oh I see that this is used throughout the place so probably I'm wrong. Ignore

Reply:
Well, you may still be correct, but there isn't much cost of doing a lock if there is no contention, but there is a big downside of not using a lock so I am going to err on the side of caution